device-server/src/main/kotlin/com/badoo/automation/deviceserver/host/HostFactory.kt

```diff
@@ -21,41 +23,44 @@ class HostFactory(
+                        remote = remoteProvider(config.host, config.user, config.publicHost),
+                        hostChecker = SimulatorHostChecker(
+                                remoteProvider(config.host, config.user, config.publicHost),
+                                wdaBundle = wdaSimulatorBundle,
+                                remoteWdaBundleRoot = REMOTE_WDA_BUNDLE_ROOT,
+                                fbsimctlVersion = fbsimctlVersion
+                        ),
+                        simulatorLimit = config.simulatorLimit,
+                        concurrentBoots = config.concurrentBoots,
+                        wdaRunnerXctest = getWdaRunnerXctest(
+                                remoteProvider(config.host, config.user, config.publicHost).isLocalhost(),
+                                wdaSimulatorBundle, REMOTE_WDA_BUNDLE_ROOT)
+                )
+                else -> DevicesNode(
+                        remoteProvider(config.host, config.user, config.publicHost),
+                        whitelistedApps = config.whitelistApps,
+                        knownDevices = config.knownDevices,
+                        uninstallApps = config.uninstallApps,
+                        wdaBundlePath = wdaDeviceBundle,
+                        remoteWdaBundleRoot = REMOTE_WDA_DEVICE_BUNDLE_ROOT,
+                        wdaRunnerXctest = getWdaRunnerXctest(
+                                remoteProvider(config.host, config.user, config.publicHost).isLocalhost(),
+                                wdaDeviceBundle, REMOTE_WDA_DEVICE_BUNDLE_ROOT),
+                        fbsimctlVersion = fbsimctlVersion
+                )
+            }
         }
-
-        return if (config.type == NodeConfig.NodeType.Simulators) {
-            SimulatorsNode(
-                remote = remote,
-                hostChecker = SimulatorHostChecker(
-                    remote,
-                    wdaBundle = wdaSimulatorBundle,
-                    remoteWdaBundleRoot = REMOTE_WDA_BUNDLE_ROOT,
-                    fbsimctlVersion = fbsimctlVersion
-                ),
-                simulatorLimit = config.simulatorLimit,
-                concurrentBoots = config.concurrentBoots,
-                wdaRunnerXctest = getWdaRunnerXctest(remote.isLocalhost(), wdaSimulatorBundle, REMOTE_WDA_BUNDLE_ROOT)
-            )
-        } else {
-            DevicesNode(
-                remote,
-                whitelistedApps = config.whitelistApps,
-                knownDevices = config.knownDevices,
-                uninstallApps = config.uninstallApps,
-                wdaBundlePath = wdaDeviceBundle,
-                remoteWdaBundleRoot = REMOTE_WDA_DEVICE_BUNDLE_ROOT,
-                wdaRunnerXctest = getWdaRunnerXctest(remote.isLocalhost(), wdaDeviceBundle, REMOTE_WDA_DEVICE_BUNDLE_ROOT),
-                fbsimctlVersion = fbsimctlVersion
-            )
+        catch (e: Exception) {
+            throw RuntimeException("Exception at ${config}", e)
```

Comment:
Maybe add more info and formatting? 
```
throw RuntimeException("Exception while creating a node from config:\n${config}", e)
```

Reply:
Sure, though I don't think it adds anything useful.